root/etc/e-smith/templates/etc/httpd/conf.d/default-virtualhost.inc/15user_settings

```diff
@@ -0,0 +1,21 @@
+#
+# 50 user_settings
+#
+
+ProxyPreserveHost On
+ProxyRequests Off
+
+RewriteEngine On
+RewriteCond %{HTTP:Upgrade} =websocket [NC]
+RewriteRule /(.*)           ws://127.0.0.1:9090/$1 [P,L]
+RewriteCond %{HTTP:Upgrade} !=websocket [NC]
+RewriteRule /(.*)           http://127.0.0.1:9090/$1 [P,L]
```

Comment:
I don't get why this block is needed, as you use the Proxy directives in the next lines.

I'm afraid mixing Rewrite* and Proxy* directives leads to configuration mistakes because they are processed with different priorities by the mod_rewrite and mod_proxy components. Can we use just _one_ approach?

I'd prefer the proxy/core one, if possible

Reply:
This is suggested by upstream: https://github.com/cockpit-project/cockpit/wiki/Proxying-Cockpit-over-Apache-with-LetsEncrypt

I'd leave it there, just to be sure :)

Reply:
Sorry I cannot find the lines below in the upstream suggestion. It's not the same code.

Reply:
It is the same code, at least diff command tells so.
I agree the rewrite rule is too wide, since the upstream code refers to a dedicated virtual host for cockpit.

Reply:
I've tryed without the `RewriteRules`, it seems it works. We can remove the rewrites and test the redirect without them